### DISCUSSION TITLE
Topaz denoise training on super-res data

### DISCUSSION THREAD

**Discussion by rpiwowarczyk** [tbepler/topaz#142] [Q&A]
I'm trying to train a new model for topaz denoise on my dataset but my raw data are in super-res (0.54A) while the motion corrected images that I want to denoise are already 2x binned (1.08A). 
Will the discrepancy affect the denoise model? Can I bin my movie stacks inside topaz (I tried topaz downsample but no luck with tif stacks).

**Comment by tbepler** [comment]
Yes, this discrepancy is likely to be an issue. You can apply a model trained on 0.54A data to 1.08A data by first upsampling the 1.08A data to 0.54A, denoising it, and then downsampling back to 1.08A, but it will be easier and the model will likely perform better if the training data and prediction data are at the same resolution.

Internally, topaz cannot bin your data before training, so you would need to first bin your movie frames some other way. I'll not this for future updates to topaz though.

**Comment by rpiwowarczyk** [comment]
Thanks! I figured I can use e2proc2d to bin the tifs and write mrcs stacks
that I could later use for training the net but it's CPU based so Fourier
transform is rather slow. Any suggestions how could I crop it using GPUs?

**Comment by tbepler** [comment]
Other than writing your own code, I'm not sure unfortunately. If you want to write your own, take a look at https://pytorch.org/docs/stable/generated/torch.nn.functional.interpolate.html#torch.nn.functional.interpolate and/or https://pytorch.org/docs/stable/generated/torch.fft.fft2.html#torch.fft.fft2 for GPU enabled downsampling!
